src/cargo/ops/cargo_new.rs

```diff
@@ -645,19 +645,37 @@ fn mk(config: &Config, opts: &MkOptions<'_>) -> CargoResult<()> {
     init_vcs(path, vcs, config)?;
     write_ignore_file(path, &ignore, vcs)?;
 
-    let (author_name, email) = discover_author(path)?;
-    let author = match (cfg.name, cfg.email, author_name, email) {
-        (Some(name), Some(email), _, _)
-        | (Some(name), None, _, Some(email))
-        | (None, Some(email), name, _)
-        | (None, None, name, Some(email)) => {
+    let (discovered_name, discovered_email) = discover_author(path);
+
+    // "Name <email>" or "Name" or "<email>" or None if neither name nor email is obtained
+    // cfg takes priority over the discovered ones
+    let author_name = match (cfg.name, discovered_name) {
+        (Some(name), _) | (_, Some(name)) => Some(name),
+        (None, None) => None,
+    };
```

Comment:
I think this and the below `match` can be expressed as `cfg.name.or(discovered_name)`

Reply:
Thank you for your feedback and guidance.